### ISSUE TITLE
Use mail.Address instead of plain strings

### ISSUE THREAD

**Issue by ivanjaros** [jordan-wright/email#129]
With simple plain strings we are unable to set the name, only the address of the recipient(s).
Use mail.Address.String() for headers, where full format works, and use mail.Address.Address for smtp's mail and rcpt methods to make it work with the smtp protocol.

**Comment by bosim** [comment]
You mean what is done in #132 ?

**Comment by ivanjaros** [comment]
no. i mean the mail address object should be used in place of string arguments, ie. every occurrence of email should be an object.

**Comment by bosim** [comment]
Make sense, however it will break existing functionality, and will not make a big difference, since the library anyway use `Address.String()` for serialization.
